src/som/interpreter/SArguments.java

```diff
@@ -35,7 +37,15 @@ public static DynamicObject getEnvironment(final Frame frame) {
   }
   
   public static ExecutionLevel getExecutionLevel(final Frame frame) {
-    return (ExecutionLevel)args(frame)[EXECUTION_LEVEL_IDX];
+    Object execLevel = args(frame)[EXECUTION_LEVEL_IDX];
+    if (execLevel.getClass() == ExecutionLevel.class) {
+      // Graal can't know that only this branch is ever taken, so, we need to help it a little
+      return (ExecutionLevel) execLevel;
+    } else {
+      // this should never happen
+      CompilerDirectives.transferToInterpreter();
+      return (ExecutionLevel) execLevel;
+    }
   }
 
```

Comment:
Just didn't get why it would be better to add explicitly the transferToInterpreter() and not a BranchProfile.